grades/models.py

```diff
@@ -223,7 +225,8 @@ def for_user_course(cls, user, course):
         """
         Returns a queryset of the exam result for an user in a course of a program
         """
-        return cls.objects.filter(user=user, course=course)
+        grades_available_runs = ExamRun.get_all_with_available_grades(course)
+        return cls.objects.filter(user=user, course=course, exam_run__in=grades_available_runs)
```

Comment:
That's fine, I agree with that. Postgres probably optimizes the subquery into a join anyway.